Rewrite the framework integration documentation (#2833)
doctrine-extensions/DoctrineExtensions: doc/frameworks/symfony.md

```yaml
services:
    # Gedmo Uploadable Extension Listener
    gedmo.listener.uploadable:
        class: Gedmo\Uploadable\UploadableListener
        tags:
            - { name: doctrine.event_listener, event: 'loadClassMetadata'}
            - { name: doctrine.event_listener, event: 'preFlush'}
            - { name: doctrine.event_listener, event: 'onFlush'}
            - { name: doctrine.event_listener, event: 'postFlush'}
        calls:
            # Uncomment the below call if using attributes, and comment the call for the annotation reader
            # - [ setAnnotationReader, [ '@gedmo.mapping.driver.attribute' ] ]
            # The `annotation_reader` service was deprecated in Symfony 6.4 and removed in Symfony 7.0
            - [ setAnnotationReader, [ '@annotation_reader' ] ]
```

## Registering Mapping Configuration

When using the [Loggable](../loggable.md), [Translatable](../translatable.md), or [Tree](../tree.md) extensions, you will
need to register the mappings for these extensions to your object managers.

> [!NOTE]

> > These extensions only provide mappings through annotations or attributes; if using annotations, you will need to ensure the deprecated [`doctrine/annotations`](https://www.doctrine-project.org/projects/annotations.html) library is installed and enabled.

Reviewer: AFAIK, the deprecation mentioned here operates on the library implementation, [not on the library itself](https://github.com/doctrine/annotations/blob/94f40ad7ecbc6931958faa8a57c48dce5da2b468/README.md?plain=1#L1-L5).

>⚠️ PHP 8 introduced [attributes](https://www.php.net/manual/en/language.attributes.overview.php), which are a native replacement for annotations. As such, this library is considered feature complete, and should receive exclusively bugfixes and security fixes.

Reviewer: I understand.
IMO, from our perspective, the important thing in our documentation is our deprecation for the annotation mapping driver, regardless of the library deprecation.
https://github.com/doctrine-extensions/DoctrineExtensions/blob/e85560ed96f977b8c29428a99222cb2ef2f0e80d/README.md?plain=1#L64-L65

Author: OK, last commit should (hopefully) make this a bit more clear across all pages I've updated so far.

Reviewer: Thank you!

                    is_bundle: false
                tree:
                    type: attribute # or annotation
                    alias: GedmoTree
                    prefix: Gedmo\Tree\Entity
                    dir: "%kernel.project_dir%/vendor/gedmo/doctrine-extensions/src/Tree/Entity"
                    is_bundle: false
```

To verify your configuration, you can use the `doctrine:mapping:info` command to make sure the entities are registered.

```sh
$ bin/console doctrine:mapping:info
 Found X mapped entities:
 [OK]   Gedmo\Loggable\Entity\LogEntry
 [OK]   Gedmo\Loggable\Entity\MappedSuperclass\AbstractLogEntry
 [OK]   Gedmo\Translatable\Entity\MappedSuperclass\AbstractPersonalTranslation
 [OK]   Gedmo\Translatable\Entity\MappedSuperclass\AbstractTranslation
 [OK]   Gedmo\Translatable\Entity\Translation
 [OK]   Gedmo\Tree\Entity\MappedSuperclass\AbstractClosure
```

## Registering Filters

### Soft Deleteable Filter

When using the [Soft Deleteable](../softdeleteable.md) extension, a filter is available which allows configuring whether
soft-deleted objects are included in query results.

> [!NOTE]

> > The Symfony bundles default filters to not being enabled, these examples will enable the filters by default.

Reviewer: The intended message here isn't clear enough to me.
Which bundles are you referring? "doctrine/doctrine-bundle" and "doctrine/mongodb-odm-bundle"?

Author: Right.  Basically the `$config->addFilter()` APIs that these configs get transformed into (and same for the Laminas config) only register the filter classes to the object managers, they don't get enabled by default.  The Symfony bundles wire extra calls to enable them globally with their respective `ManagerConfigurator` classes.

Reviewer: ```suggestion
> The default configuration in the Symfony bundles don't enable the filters. These examples show how to globally enable them.
```

Author: Updated, thanks!
